cli: change jemalloc defaults

RaduBerinde opened the pull request:
This commit changes the jemalloc configuration to reduce Cgo total
spikes. This configuration has shown to reduce spikes without any
noticeable impact to performance.

Epic: none
Release note: None

Full experiment results (note the multiple tabs): https://docs.google.com/spreadsheets/d/1Vs9hx0wytkySIvW3DRXI4MXC49b_9Bgkk_IPsoKFH6M/edit?gid=0#gid=0

This is a before/after of sysbench with 3 nodes and 800M rows:
<img width="1729" alt="image" src="https://github.com/user-attachments/assets/3362bee7-3841-4a2e-843f-9fe683057ce5" />

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139743)
<!-- Reviewable:end -->

RaduBerinde commented:
Any thoughts on backporting to 25.1?  I would be in favor, it seems like a row risk change.

sumeerbhola reviewed (approved):
Great! I'm in favor of backporting.

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139743#-:-OHP_xZS2YWCSDjfVmlq:b-qqjbms)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jbowens, @nicktrav, and @RaduBerinde)

___
*[`pkg/cli/start_jemalloc.go` line 30 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/139743#-OHP_O7bA6iPYCNopPLw:-OHP_O7bA6iPYCNopPLx:bjt78gn) ([raw file](https://github.com/cockroachdb/cockroach/blob/9f31dcdf441df98099fa5e782bcfa737744aec23/pkg/cli/start_jemalloc.go#L30)):*
> ```Go
> // // MALLOC_CONF env var. From the jemalloc man page:
> // //   The string specified via --with-malloc-conf, the string pointed to by
> // //   the global variable malloc_conf, the “name” of the file referenced by
> ```

Is this global variable the same as the `je_` prefixed one below?

Gemini claimed the following -- wondering whether to believe it:
**`malloc_conf` variable (Deprecated/Rarely Used)**

In some (older) versions of jemalloc, there was a global `const char *` variable called `malloc_conf`. You might see it referenced in older documentation or code.

**This method is generally deprecated and should be avoided in favor of `mallctl*`**


<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Yes, it's the same, our build is prepending `je_` to all symbols. The latest man page (https://jemalloc.net/jemalloc.3.html, section TUNING) doesn't mention anything about deprecation. Some of the settings are hard or impossible to apply via `mallctl` after initialization.

RaduBerinde commented:
Hm, I see some "Invalid conf pair: muzzy_delay_ms:0" messages in roachtests. I will investigate. Though this is 0 by default in our version anyway so it'd be ok to remove.

RaduBerinde commented:
Oh, it's just a typo, it should be `decay`

RaduBerinde committed:
cli: change jemalloc defaults

This commit changes the jemalloc configuration to reduce Cgo total
spikes. This configuration has shown to reduce spikes without any
noticeable impact to performance.

Epic: none
Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139743#-:-OHQUxyr4XBb-QwFoO_y:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jbowens, @nicktrav, and @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
